### ISSUE TITLE
NativeMemoryProfiler exception

### ISSUE THREAD

**Issue by WojciechNagorski** [dotnet/BenchmarkDotNet#1235]
@dotMorten has reproduced an exception in `NativeMemoryProfiler`. 
There is https://twitter.com/dotMorten/status/1166842655348670466 

At first, he got an error:
![image](https://user-images.githubusercontent.com/17333903/63920870-04e0bb80-ca42-11e9-9645-233bc701eaae.png)

Then:
> Setting ArtifactsPath="./" in my manual config seems to work (null check missing there?). Now it runs for a lot longer and creates .etl files, then goes boom here:

![image](https://user-images.githubusercontent.com/17333903/63920989-3eb1c200-ca42-11e9-97ea-6d646b5dbcb1.png)

**Comment by adamsitnik** [comment]
@wojtpl2 are you able to repro it?

**Comment by wojtpl2** [comment]
@adamsitnik No, I'm not. 
@dotMorten Could you share some reproduction? 

I have some ideas:
- Copy of native DLL
Related with https://github.com/dotnet/BenchmarkDotNet/issues/946 . I know good solution for it. Instead of using `AfterBuild`, use this construction: https://github.com/wojtpl2/NativeMemorySample/blob/master/NativeDllWraper/NativeDllWraper.csproj#L6-L9. I checked that this solution worked even for the directory. I'm writing a blog post about it. 
- Working without admin?
- Use this `NativeMemoryProfiler` with another profiler which uses `EtwProfiler`.

**Comment by wojtpl2** [comment]
the first exception occurred in this line:
https://github.com/dotnet/BenchmarkDotNet/blob/e92abf845ac916e449bf9d8ccf1d872204951bec/src/BenchmarkDotNet.Diagnostics.Windows/Sessions.cs#L147
It is possible only if `details.Config.ArtifactsPath` is null. I need time to understand when this can be null.
@dotMorten If you are not able to share reproduction, please share only your configuration. I mean `csproj` and benchmark configuration.

**Comment by dotMorten** [comment]
I should be able to share something once I get into the office in a bit. The second crash only occurs for some tests, and the etl file was huuuuge at that point (over 8gb).
Only see two places where division by error can occur, but not sure why those values would be zero at that point.

**Comment by wojtpl2** [comment]
I've just pushed nicer error handling. Division by zero is possible only when *.etl file doesn't contain events from BenchmarkDotNet Engine. To be honest, I don't know yet why it works that way, but I will find this bug 😄

**Comment by dotMorten** [comment]
Below is a repro for the artifacts path. I'll get back to you with the other issue, once I zero in on it (the one that fails is overly complicated so trying to dumb it down):

You'll need this package reference:
```xml
    <PackageReference Include="Esri.ArcGISRuntime.WPF">
      <Version>100.6.0</Version>
    </PackageReference>
```
Repro app code:
```cs
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Configs;
using BenchmarkDotNet.Jobs;
using BenchmarkDotNet.Running;
using BenchmarkDotNet.Toolchains.CsProj;
using Esri.ArcGISRuntime.Geometry;
using System;
using System.Linq;

namespace BDN_Repro
{
    class Program
    {
        static void Main(string[] args)
        {
            var config = new MyConfig();
            var summary = BenchmarkRunner.Run(typeof(GeometryBuilderBenchmarks), config);
        }
    }
    public class GeometryBuilderBenchmarks
    {
        MapPointBuilder pointbuilder;
        [GlobalSetup]
        public void Setup()
        {
            pointbuilder = new MapPointBuilder(123, 456, SpatialReferences.Wgs84);
        }

        [Benchmark]
        public double MapPointBuilder_InteropOverhead_GetSetXY()
        {
            return ((pointbuilder.X += 1) + (pointbuilder.Y += 1));
        }
    }

    public class MyConfig : ManualConfig
    {
        private Job baseJob;
        public MyConfig()
        {
            baseJob = Job.MediumRun.With(CsProjClassicNetToolchain.Net48);


            Add(DefaultConfig.Instance.GetLoggers().ToArray());
            Add(DefaultConfig.Instance.GetExporters().ToArray());
            Add(DefaultConfig.Instance.GetColumnProviders().ToArray());
            Add(BenchmarkDotNet.Diagnosers.MemoryDiagnoser.Default);
            Add(new BenchmarkDotNet.Diagnostics.Windows.NativeMemoryProfiler());
            // ArtifactsPath = "./"; //This fixes the crash
        }
    }
}

```

**Comment by wojtpl2** [comment]
@dotMorten Thanks for this reproduction. I've just pushed fix. 

The problem was not in `NativeMemoryProfiler`.

When you run benchmark with your own configuration in this way:
```c# 
        static void Main(string[] args)
        {
            var config = new MyConfig();
            var summary = BenchmarkRunner.Run(typeof(GeometryBuilderBenchmarks), config);
        }
```
Then resulted ArtifactsPath is null. 

I found information in the code that in this case the default path should be used.
https://github.com/dotnet/BenchmarkDotNet/blob/3bd18fc1fd84c70eedde1ced821df8777c710ec2/src/BenchmarkDotNet/Validators/ConfigValidator.cs#L41-L43

I check that the null value in ArtifactsPath causes an exception for all EtwProfiler, for instance, the configuration below raises the same exception:
```C#
        public class MyConfig : ManualConfig
        {
            private Job baseJob;
            public MyConfig()
            {
                baseJob = Job.MediumRun.With(CsProjClassicNetToolchain.Net461);


                Add(DefaultConfig.Instance.GetLoggers().ToArray());
                Add(DefaultConfig.Instance.GetExporters().ToArray());
                Add(DefaultConfig.Instance.GetColumnProviders().ToArray());
                Add(BenchmarkDotNet.Diagnosers.MemoryDiagnoser.Default);
                Add(new BenchmarkDotNet.Diagnostics.Windows.EtwProfiler()); // <-- Use EtwProfiler instead of NativeMemoryProfiler
            }
        }
```

**Comment by dotMorten** [comment]
@wojtpl2 I think what I'm seeing with the second crash is because the test threw an exception in the background thread.

**Comment by wojtpl2** [comment]
@adamsitnik @dotMorten  I think that PR #1237 is ready to review and this issue is ready to close. 

I checked that exception in a background thread stops benchmark and changes from #1237 shows the message: 
![image](https://user-images.githubusercontent.com/17333903/64102576-8dc46380-cd70-11e9-914f-0a1e09add9a2.png)

**Comment by adamsitnik** [comment]
Fixed in #1237 a while ago, but we forgot to close the issue. So I am closing it now.
